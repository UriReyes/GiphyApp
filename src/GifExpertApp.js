import React, { useState } from "react";
import { GifFooter } from "./components/GifFooter";

import { GifGrid } from "./components/GifGrid";
import { GifHeader } from "./components/GifHeader";

const GifExpertApp = () => {
  //const categories = ["Naruto", "Dragon Ball", "PokÃ©mon"];
  const [categories, setCategories] = useState(["DOOM Eternal"]);

  //   const handleAdd = (e) => setCategories([...categories, "Baki"]);

  return (
    <>
      <GifHeader setCategories={setCategories} />
      <div>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </div>
      <GifFooter />
    </>
  );
};

export default GifExpertApp;
