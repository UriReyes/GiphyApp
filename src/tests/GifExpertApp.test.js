import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Test for <GifExpertApp/>", () => {
  test("should do match with the snapshot", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show a list of categories", () => {
    const categories = ["Got", "Doom Eternal"];
    const wrapper = shallow(<GifExpertApp defaultCategory={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
