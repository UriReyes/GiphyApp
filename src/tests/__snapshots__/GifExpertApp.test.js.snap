// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <GifExpertApp/> should do match with the snapshot 1`] = `
<Fragment>
  <GifHeader
    setCategories={[Function]}
  />
  <div>
    <div
      className="title-app text-center p-5"
    >
      Aún no has buscado ningún Gif...
    </div>
  </div>
  <GifFooter />
</Fragment>
`;

exports[`Test for <GifExpertApp/> should show a list of categories 1`] = `
<Fragment>
  <GifHeader
    setCategories={[Function]}
  />
  <div>
    <GifGrid
      category="Got"
      key="Got"
    />
    <GifGrid
      category="Doom Eternal"
      key="Doom Eternal"
    />
  </div>
  <GifFooter />
</Fragment>
`;
